iroh-rpc-client/src/client.rs

```diff
@@ -1,26 +1,76 @@
+use std::collections::HashMap;
 use std::net::SocketAddr;
 
 use anyhow::{Context, Result};
+use async_stream::stream;
+use futures::{Stream, StreamExt};
 
+use crate::gateway::GatewayClient;
 use crate::network::P2pClient;
+use crate::status::ServiceStatus;
 use crate::store::StoreClient;
 
 #[derive(Debug, Clone)]
 pub struct Client {
+    pub gateway: GatewayClient,
     pub p2p: P2pClient,
     pub store: StoreClient,
 }
 
 impl Client {
     pub async fn new(cfg: &RpcClientConfig) -> Result<Self> {
+        let gateway = GatewayClient::new(cfg.gateway_addr)
+            .await
+            .context("Could not create gateway rpc client")?;
         let p2p = P2pClient::new(cfg.p2p_addr)
             .await
             .context("Could not create p2p rpc client")?;
         let store = StoreClient::new(cfg.store_addr)
             .await
             .context("Could not create store rpc client")?;
 
-        Ok(Client { p2p, store })
+        Ok(Client {
+            gateway,
+            p2p,
+            store,
+        })
+    }
+
+    pub async fn check(&self) -> HashMap<String, ServiceStatus> {
+        let mut s = HashMap::default();
+        s.insert("store".into(), self.store.check().await);
+        s.insert("p2p".into(), self.p2p.check().await);
+        s.insert("gateway".into(), self.gateway.check().await);
+        s
+    }
+
+    pub async fn watch(self) -> impl Stream<Item = HashMap<String, ServiceStatus>> {
+        stream! {
+            let mut statuses = self.check().await;
+            yield statuses.clone();
+            let store_status = self.store.watch().await;
+            futures::pin_mut!(store_status);
```

Comment:
I would suggest using `tokio::pin!`, given that we use tokio for the stream stuff, (they should do roughly the same, just to be on the safe side.